src/engine/mega-macro-action-bar-engine.lua

```diff
@@ -401,8 +401,12 @@ function MegaMacroActionBarEngine.OnUpdate(elapsed)
 
 	iterator(function(button)
 		local action = button:GetAttribute("action") or button.action
+		local macroName = GetActionText(action)
+		local macroCode = GetMacroBody(macroName)
         local type, actionArg1 = GetActionInfo(action)
-        local macroId = type == "macro" and MegaMacroEngine.GetMacroIdFromIndex(actionArg1)
```

Comment:
I'd keep it as is for now, hoping that Blizzard will fix the GetActionInfo bug. In the event of a resolution, this workaround should no longer be necessary.

(Note: This workaround uses macro names to identify macros which are not 100% reliable since macros can share a name.)